jwt/jwks_client.py

```diff
@@ -8,8 +9,12 @@
 
 
 class PyJWKClient:
-    def __init__(self, uri: str):
+    def __init__(self, uri: str, cache_keys: bool = True):
         self.uri = uri
+        if cache_keys:
+            # Cache signing keys
+            # Ignore mypy (https://github.com/python/mypy/issues/2427)
+            self.get_signing_key = lru_cache(maxsize=16)(self.get_signing_key)  # type: ignore
```

Comment:
@makusu2 I think we should also make maxsize configurable here.

Reply:
Good call, done :+1: